mindocr/data/transforms/svtr_transform.py

```diff
@@ -242,59 +239,66 @@ def __init__(self, factor=4, base_size=(128, 512)):
             factor: the decayed factor from base size, factor=4 keeps target scale by default.
             base_size: base size the build the bottom layer of pyramid
         """
-        if isinstance(factor, numbers.Number):
-            self.factor = round(sample_uniform(0, factor))
-        elif isinstance(factor, (tuple, list)) and len(factor) == 2:
-            self.factor = round(sample_uniform(factor[0], factor[1]))
-        else:
-            raise Exception("factor must be number or list with length 2")
         # assert factor is valid
+        self.factor = factor
         self.base_h, self.base_w = base_size[:2]
 
     def __call__(self, img):
-        if self.factor == 0:
+        if isinstance(self.factor, numbers.Number):
+            factor = round(sample_uniform(0, self.factor))
+        elif isinstance(self.factor, (tuple, list)) and len(self.factor) == 2:
+            factor = round(sample_uniform(self.factor[0], self.factor[1]))
+        else:
+            raise RuntimeError("factor must be number or list with length 2")
```

Comment:
How about using lambda functions inside `__init__` to remove data type checking inside `__call__`? Just like here: https://github.com/mindspore-lab/mindocr/pull/430/commits/af54a3ac187878869e28041f5a58a8380eefc8e8

Reply:
Yes, this should reduce some computation time, can be replace by #430 once it is merged